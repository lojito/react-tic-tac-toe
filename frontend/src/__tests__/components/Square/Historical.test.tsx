import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import SquareHistorical from "../../../components/Square/Historical";

describe("SquareHistorical", () => {
  const renderSquare = (imagePath = "default") => {
    const utils = render(
      <SquareHistorical
        disabled
        id="100"
        imagePath={imagePath}
        className="win"
      />
    );
    const squareButton = screen.getByRole("button", { name: "historical" });

    return { ...utils, squareButton };
  };

  it("should render correctly", () => {
    const { asFragment } = renderSquare();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should apply the props", () => {
    const imagePath = "soccer/6";

    const { squareButton } = renderSquare(imagePath);

    expect(squareButton).toBeDisabled();
    expect(squareButton).toHaveAttribute("id", "100");
    expect(squareButton).toHaveAttribute("class", "win");
    expect(squareButton).toHaveStyle({ backgroundImage: /soccer\/6/ });
  });
});
