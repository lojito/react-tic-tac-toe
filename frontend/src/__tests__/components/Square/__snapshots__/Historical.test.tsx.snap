// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareHistorical should render correctly 1`] = `
<DocumentFragment>
  <button
    aria-label="historical"
    class="win"
    disabled=""
    id="100"
  />
</DocumentFragment>
`;
